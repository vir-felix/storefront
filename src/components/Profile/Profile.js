import React, { Component } from "react";
import PropTypes from "prop-types";
import Typography from "material-ui/Typography";

export default class Profile extends Component {
  static propTypes = {
    viewer: PropTypes.object
  };

  static defaultProps = {
    viewer: {}
  };

  render() {
    const { viewer: { name } } = this.props;
    return <Typography variant="subheading">Hello {name}</Typography>;
  }
}
