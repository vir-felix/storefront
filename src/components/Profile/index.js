import React, { Component } from "react";
import PropTypes from "prop-types";
import { Query } from "react-apollo";
import gql from "graphql-tag";

const GET_VIEWER = gql`
  {
    viewer {
      name
      emailRecords {
        address
      }
    }
  }
`;

class Profile extends Component {
  static propTypes = {
    viewer: PropTypes.object
  }

  render() {
    const { viewer: { name } } = this.props;
    return (
      <h1>Hello {name}</h1>
    );
  }
}

export default () => (
  <Query query={GET_VIEWER}>
    {({ loading, error, data }) => {
      if (loading) return <p>Loading...</p>;
      if (error) return <p>Error</p>;

      const { viewer } = data;
      return (
        <Profile viewer={viewer} />
      );
    }}
  </Query>
);
