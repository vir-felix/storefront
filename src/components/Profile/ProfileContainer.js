import React from "react";
import { Query } from "react-apollo";
import viewerQuery from "./viewer.gql";
import Profile from "./Profile";

export default () => (
  <Query query={viewerQuery}>
    {({ loading, error, data }) => {
      if (loading) return <p>Loading...</p>;
      if (error) return <p>Error</p>;

      const { viewer } = data;
      return (
        <Profile viewer={viewer} />
      );
    }}
  </Query>
);
