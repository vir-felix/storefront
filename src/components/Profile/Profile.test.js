import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Profile from "./Profile";

test("basic snapshot", () => {
  const component = renderer.create(shallow(<Profile />).get(3));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
