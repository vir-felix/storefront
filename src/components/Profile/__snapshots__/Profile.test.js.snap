// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
<h3
  className="MuiTypography-root-1 MuiTypography-subheading-8"
>
  Hello 
  Test User
</h3>
`;
